packages/js/product-editor/src/blocks/product-fields/product-list/edit.tsx

```diff
@@ -0,0 +1,282 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
+import { resolveSelect } from '@wordpress/data';
+import {
+	createElement,
+	useContext,
+	useEffect,
+	useState,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { external, closeSmall } from '@wordpress/icons';
+import { useWooBlockProps } from '@woocommerce/block-templates';
+import { CurrencyContext } from '@woocommerce/currency';
+import { PRODUCTS_STORE_NAME, Product } from '@woocommerce/data';
+import { getNewPath } from '@woocommerce/navigation';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import {
+	AddProductsModal,
+	getProductImageStyle,
+} from '../../../components/add-products-modal';
+import { ProductEditorBlockEditProps } from '../../../types';
+import { Shirt, Pants, Glasses } from './images';
+import { UploadsBlockAttributes } from './types';
+import {
+	getProductStockStatus,
+	getProductStockStatusClass,
+} from '../../../utils';
+
+export function Edit( {
+	attributes,
+	context: { postType },
+}: ProductEditorBlockEditProps< UploadsBlockAttributes > ) {
+	const { property } = attributes;
+	const blockProps = useWooBlockProps( attributes );
+	const [ openAddProductsModal, setOpenAddProductsModal ] = useState( false );
+	const [ isLoading, setIsLoading ] = useState( false );
+	const [ groupedProductIds, setGroupedProductIds ] = useEntityProp<
+		number[]
+	>( 'postType', postType, property );
+	const [ groupedProducts, setGroupedProducts ] = useState< Product[] >( [] );
+	const { formatAmount } = useContext( CurrencyContext );
+
+	useEffect(
```

Comment:
How come you used a `useEffect` and `resolveSelect` here?
Wouldn't a `useSelect` actually be easier here, given it is not being triggered upon an event.
`useSelect` also accepts the dependency array, so you can still add the `groupedProductIds` there.